ste/mgr-JobMgr.go

```diff
@@ -575,9 +575,8 @@ func (jm *jobMgr) reportJobPartDoneHandler() {
 			fmt.Println("reportJobPartDoneHandler done called")
 			return
 
-		default:
+		case partProgressInfo := <-jm.jobPartProgress:
```

Comment:
to avoid processing cancel before jobProgress messages sitting in the channel, we should do like

```
	for {
		select {
		case partProgressInfo := <-jm.jobPartProgress:
		default:
		    // If No jobPartProgress message, then check if we need to exit.
            case <-jm.reportCancelCh:
                fmt.Println("reportJobPartDoneHandler done called")
                return
            case partProgressInfo := <-jm.jobPartProgress:
            // Fall through.
            // The double case pattern ensure we always check jm.jobPartProgress before jm.reportCancelCh,
            // so if we have a channel cancel pending and some jobPartProgress messages we process the
            // jobPartProgress messages before the cancel.
        }

        // Handling code here.
    }
```

Reply:
It'll be good if we can catch issues where jobPartProgress messages are received after channel is closed.
f.e. to indicate channel closure we can close jm.reportCancelCh and then in ReportJobPartDone we can check that reportCancelCh should not be closed.
Makese sense?

Reply:
for {
		select {
		case partProgressInfo := <-jm.jobPartProgress:
                case <-jm.reportCancelCh:
               }
}